spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/GcpPubSubAutoConfiguration.java

```diff
@@ -153,6 +153,12 @@ public PubSubPublisherTemplate pubSubPublisherTemplate(PublisherFactory publishe
 		return pubSubPublisherTemplate;
 	}
 
+	@Bean
+	@ConditionalOnMissingBean(name = "pubSubAsynchronousPullExecutor")
```

Comment:
When writing this I was in doubt what to do. Normally I would also use an `Optional` or `ObjectProvider`. But somehow I thought that wouldn't be in line with how this autoconfiguration class worked. Looking at it again now, I see many usages of `ObjectProvider`... 
So, I applied your suggestion, but instead of using an `Optional` I used a `ObjectProvider`, as it is used all over the place, instead of `Optional`. So, its more in line with the rest of the class.